### PR TITLE
Create game loop

### PR DESCRIPTION
**Pull request by ikarabulut**
Added `GameSettings` to be responsible for the game setting. This class is injected into `Game` which takes the evolutions number from game settings and uses that to loop through the generateNextBoard method n(evolutions) times.

### CODE REVIEW

File: app/src/main/java/com/ikarabulut/Game.java [hunk 1]
```diff
@@ -2,9 +2,19 @@
...
     private Board board;
+    private GameSettings gameSettings;
 
-    public Game(Board theBoard) {
+    public Game(Board theBoard, GameSettings theGameSettings) {
         board = theBoard;
+        gameSettings = theGameSettings;
+    }
+
+    public void startEvolutionLoops() {
```
Comment (riccjohn) [inline app/src/main/java/com/ikarabulut/Game.java:12]: You can probably just call this 'start'

File: app/src/test/java/com/ikarabulut/GameSettingsTest.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.ikarabulut;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class GameSettingsTest {
```
Comment (riccjohn) [inline app/src/test/java/com/ikarabulut/GameSettingsTest.java:5]: Empty? :(

File: app/src/main/java/com/ikarabulut/Game.java [hunk 3]
```diff
@@ -2,9 +2,19 @@
...
+        int numOfEvolutions = gameSettings.getEvolutions();
+
+        for (int i=0; i<numOfEvolutions; i++) {
```
Comment (riccjohn) [inline app/src/main/java/com/ikarabulut/Game.java:15]: I know we don't have the display part done yet, but how are you envisioning this working? Would you iterate over the number of evolutions and then print each evolution to the screen?

**Review by riccjohn — APPROVED** [review]
Approved! Just have some minor changes/questions.

File: app/src/test/java/com/ikarabulut/GameSettingsTest.java (hunk 2 above)
Comment (ikarabulut) [reply app/src/test/java/com/ikarabulut/GameSettingsTest.java:5]: I wasnt sure if tests would be needed for getters. Its a pretty straightforward method

File: app/src/main/java/com/ikarabulut/Game.java (hunk 3 above)
Comment (ikarabulut) [reply app/src/main/java/com/ikarabulut/Game.java:15]: Idk yet, that i what I am going to work on now.

I kinda wish I had a better understanding of Outside-in development, I thought I did but I was wrong, and now Im working in a pretty unorganized codebase.

**Pull request merged**